x-pack/plugin/sql/src/main/java/org/elasticsearch/xpack/sql/expression/function/aggregate/PercentileRanks.java

```diff
@@ -5,35 +5,64 @@
  */
 package org.elasticsearch.xpack.sql.expression.function.aggregate;
 
+import org.elasticsearch.search.aggregations.metrics.PercentilesConfig;
 import org.elasticsearch.xpack.ql.expression.Expression;
 import org.elasticsearch.xpack.ql.tree.NodeInfo;
 import org.elasticsearch.xpack.ql.tree.Source;
 
 import java.util.List;
+import java.util.Objects;
 
 public class PercentileRanks extends CompoundNumericAggregate {
```

Comment:
Considering this file and Percentiles have grew in size and have two properties plus the associated equals/hashCode is worth extracting the functionlity into a common base class that extends CompoundNumericAggregate and just like their function equivalent, the actual implementations only inherit and declare the values/fields as a synonim to the internal expressions list and handle the info and replaceChildren methods.

Reply:
While implementing this suggested change, found the following:

```java
    public InnerAggregate(Source source, AggregateFunction inner, CompoundAggregate outer, Expression innerKey) {
        super(source, outer.field(), outer.arguments());
```

Where the `outer.arguments()` delegates to `children()` which already contains the `field()`, so the InnerAggregate will have two `field`s in the `children` list. Will address it separately, seemingly it does not cause any issues.